### PR TITLE
Add TypeRepr.memberInfo, Symbol.tpe, Symbol.typeParams to reflection API

### PR DESCRIPTION
**Pull request by pikinier20**

### CODE REVIEW

**Comment by nicolasstucki** [comment]
Part of #13371

File: compiler/src/scala/quoted/runtime/impl/QuotesImpl.scala [hunk 1]
```diff
@@ -2610,6 +2612,8 @@ class QuotesImpl private (using val ctx: Context) extends Quotes, QuoteUnpickler
         def companionClass: Symbol = self.denot.companionClass
         def companionModule: Symbol = self.denot.companionModule
         def children: List[Symbol] = self.denot.children
+        def tpe: TypeRepr = self.typeRef
```
Comment (nicolasstucki) [inline compiler/src/scala/quoted/runtime/impl/QuotesImpl.scala:2615]: This was in #9707

**Review by nicolasstucki — CHANGES_REQUESTED** [review]
Each method should be added in one PR and include some tests

**Comment by pikinier20** [comment]
The problem was solved by #14892

**Pull request closed without merging**